 <launch>
<!--Input topic arguments-->
<!--<arg name="joints_topic_name" default="/omni1_joint_states"/> -->
<arg name="button_topic_name" default="/omni1_button"/> 
<arg name="lock_state_topic" default="/omni1_lock_state"/> 
<!--Output topics arguments-->
<arg name="marker_topic_name" default="/marker_visualization"/> 
<!--<arg name="marker_trans_topic_name" default="/marker_transform"/> -->
<!--This robot frame needs to be set as the reference frame that is tied to the omni -->  
<!--<arg name="robot_reference_frame"          default="wrist_ft_tool_link"/> -->
<arg name="robot_reference_frame"          default="base_footprint"/> 
<!--This name needs to be set as the omni frame-->  
<arg name="HD_frame_name"          default="omni_rotation"/> 
<!--This name needs to be set as the End-effector of the robot-->
<arg name="end_effector_frame"          default="wrist_ft_tool_link"/>
<!--This name needs to be set as the transformation frame of the virtual marker-->
<arg name="virtual_marker_name"          default="virtual_marker"/>
<!--Scaling coefficient of omni workspace to robot-->
<arg name="scale_marker_deviation"          default="0.01"/>
<arg name="lambda_min" default="0.01"/>
<arg name="lambda_max" default="0.45"/>



  <node pkg="haptic_device_2_marker" name="omni_2_marker" type="omni_2_marker" output="screen">
    <!--Input topic-->
    <!--<param name="joint_state_topic_name" value="$(arg joints_topic_name)"/>-->
    <param name="button_event_topic_name" value="$(arg button_topic_name)"/>

    <!--Output topics-->
    <param name="marker_topic_name" value="$(arg marker_topic_name)"/>
    <!--<param name="marker_trans_topic_name" value="$(arg marker_trans_topic_name)"/>-->

    <param name="robot_reference_frame"     value="$(arg robot_reference_frame)"/>
    <param name="HD_frame_name"     value="$(arg HD_frame_name)"/>
    <param name="ee_frame_name"     value="$(arg end_effector_frame)"/>
    <param name="virtual_marker_name"     value="$(arg virtual_marker_name)"/>

    <param name="scale_marker_deviation"     value="$(arg scale_marker_deviation)"/>
    <param name="lambda_min"        value="$(arg lambda_min)"/>
    <param name="lambda_max"        value="$(arg lambda_max)"/>

  </node>
 </launch>